src/mud/examples/poisson.py

```diff
@@ -419,14 +452,14 @@ def run_2d_poisson_trials(
             e_r = mud_prob.expected_ratio()
             ax2.text(x, y2, rf"$\mathbb{{E}}(r) = {e_r:0.4}$", fontsize=18)
         ax2.legend()
-
-        save_figure(
-            f"solution_n{N}",
-            save_path,
-            close_fig=close_fig,
-            dpi=dpi,
-            bbox_inches="tight",
-        )
+        if save_path:
```

Comment:
Should be 
`if save_path is not None:`
See [PEP 8](https://peps.python.org/pep-0008/)